server/src/jobs/protocols.cpp

```diff
@@ -34,87 +34,112 @@ namespace fty::job {
 enum class Type
 {
     Powercom = 1,
-    Xml      = 2,
-    Snmp     = 3,
+    Snmp     = 2,
+    Xml      = 3,
 };
 
 // =====================================================================================================================
 
 inline std::ostream& operator<<(std::ostream& ss, Type type)
 {
     switch (type) {
-        case Type::Snmp:
-            ss << "Snmp";
+        case Type::Powercom:
+            ss << "GenApi";
             break;
         case Type::Xml:
             ss << "Xml";
             break;
-        case Type::Powercom:
-            ss << "GenApi";
+        case Type::Snmp:
+            ss << "Snmp";
             break;
+		default:
+			ss << "protocol-type-unknown";
     }
     return ss;
 }
 
 // =====================================================================================================================
 
-
 void Protocols::run(const commands::protocols::In& in, commands::protocols::Out& out)
 {
     if (in.address == "__fake__") {
-        out.setValue({"nut_snmp", "nut_xml_pdc"});
+        logInfo("{} address (test)", in.address);
+        auto& x = out.append();
+        x.protocol = "nut_snmp";
+        x.port = 1234;
+        x.reachable = true;
+        auto& y = out.append();
+        y.protocol = "nut_xml_pdc";
+        y.port = 4321;
+        y.reachable = false;
         return;
     }
 
     if (!available(in.address)) {
         throw Error("Host is not available: {}", in.address.value());
     }
 
-    std::vector<Type> protocols;
-
-    if (auto res = tryXmlPdc(in)) {
-        protocols.emplace_back(Type::Xml);
-        log_info("Found XML device");
-    } else {
-        log_info("Skipped xml_pdc, reason: %s", res.error().c_str());
-    }
-
-    if (auto res = trySnmp(in)) {
-        protocols.emplace_back(Type::Snmp);
-        log_info("Found SNMP device");
-    } else {
-        log_info("Skipped snmp, reason: %s", res.error().c_str());
-    }
-
-    if (auto res = tryPowercom(in)) {
-        protocols.emplace_back(Type::Powercom);
-        log_info("Found Powercon device");
-    } else {
-        log_info("Skipped GenApi, reason: %s", res.error().c_str());
-    }
-
-    sortProtocols(protocols);
-
-    for (const auto& prot : protocols) {
-        switch (prot) {
-            case Type::Snmp:
-                out.append("nut_snmp");
+    // supported protocols, with defaults
+    struct {
+        Type protocol;
+        std::string protocolStr;
+        uint16_t port;
+    } tries[] = {
+        {Type::Powercom, "nut_powercom", 443},
```

Comment:
I think xml must be prioritized than snmp

Reply:
done. the preference order is: powercom, xml, snmp